tutorials/mesh_generation/4_tree_mesh.py

```diff
@@ -0,0 +1,247 @@
+"""
+Tree Meshes
+===========
+
+Compared to tensor meshes, tree meshes are able to provide higher levels
+of discretization in certain regions while reducing the total number of
+cells. Tree meshes belong to the class (:class:`~discretize.TreeMesh`).
+Tree meshes can be defined in 2 or 3 dimensions. Here we demonstrate:
+
+    - How to create basic tree meshes in 2D and 3D
+    - Strategies for local mesh refinement
+    - How to plot tree meshes
+    - How to extract properties from tree meshes
+
+To create a tree mesh, we first define the base tensor mesh (a mesh
+comprised entirely of the smallest cells). Next we choose the level of
+discretization around certain points or within certain regions. When
+creating tree meshes, we must remember certain rules:
+
+    - The number of base mesh cells in x, y and z must all be powers of 2
+    - We cannot refine the mesh to create cells smaller than those defining the base mesh
+    - The range of cell sizes in the tree mesh depends on the number of base mesh cells in x, y and z
+    
+
+"""
+
+###############################################
+#
+# Import Packages
+# ---------------
+#
+# Here we import the packages required for this tutorial.
+#
+
+from discretize import TreeMesh
+from discretize.utils import matutils, meshutils
```

Comment:
rather than importing the util modules, for clarify it is a bit simpler to do
```
from discretize.utils import mkvc, refine_tree_xyz
```

Reply:
This also allows us to change the folder structure after the fact (which has happened a few times already).

Reply:
```suggestion
from discretize.utils import mkvc, regine_tree_xyz
```